tests/src/testadaptivestepsizing.cpp

```diff
@@ -125,53 +129,38 @@ auto KLShellAndAdaptiveStepSizing(const PathFollowingType& pft, const std::vecto
     }
   });
 
-  auto sparseAssembler = SparseFlatAssembler(fes, dirichletValues);
+  auto sparseAssembler = makeSparseFlatAssembler(fes, dirichletValues);
 
   Eigen::VectorXd d;
   d.setZero(basis.flat().size());
   double lambda = 0.0;
 
-  auto req = FERequirements().addAffordance(Ikarus::AffordanceCollections::elastoStatics);
-
-  auto energyFunction = [&](auto&& disp_, auto&& lambdaLocal) -> auto& {
-    req.insertGlobalSolution(Ikarus::FESolutions::displacement, disp_)
-        .insertParameter(Ikarus::FEParameter::loadfactor, lambdaLocal);
-    return sparseAssembler.getScalar(req);
-  };
-
-  auto residualFunction = [&](auto&& disp_, auto&& lambdaLocal) -> auto& {
-    req.insertGlobalSolution(Ikarus::FESolutions::displacement, disp_)
-        .insertParameter(Ikarus::FEParameter::loadfactor, lambdaLocal);
-    return sparseAssembler.getVector(req);
-  };
-
-  auto KFunction = [&](auto&& disp_, auto&& lambdaLocal) -> auto& {
-    req.insertGlobalSolution(Ikarus::FESolutions::displacement, disp_)
-        .insertParameter(Ikarus::FEParameter::loadfactor, lambdaLocal);
-    return sparseAssembler.getMatrix(req);
-  };
-
-  auto nonLinOpFull =
-      Ikarus::NonLinearOperator(functions(energyFunction, residualFunction, KFunction), parameter(d, lambda));
+  auto req = fes[0].createRequirement();
+  req.insertGlobalSolution(d).insertParameter(lambda);
+  sparseAssembler->bind(req, Ikarus::AffordanceCollections::elastoStatics);
 
-  auto nonLinOp = nonLinOpFull.template subOperator<1, 2>();
-  nonLinOp.updateAll();
   auto linSolver = LinearSolver(SolverTypeTag::sd_SimplicialLDLT);
 
   int loadSteps = 6;
 
-  auto nr   = Ikarus::makeNewtonRaphsonWithSubsidiaryFunction(nonLinOp, std::move(linSolver));
+  auto nrSettings = NewtonRaphsonWithSubsidiaryFunctionConfig<decltype(linSolver)>{.linearSolver = linSolver};
```

Comment:
`nrSettings ` is meant for `NRSettings` isn't it. May be `nrConfig` is a better name?